cibyl/sources/zuuld/git_api.py

```diff
@@ -0,0 +1,216 @@
+                    temp_path = result.path.split(".git")[0]
+                    if temp_path.startswith("/"):
+                        temp_path = temp_path.lstrip("/")
+                else:
+                    if result.path.startswith("/"):
+                        temp_path = result.path.lstrip("/")
+
+                self.path = repo['repo_name'] = temp_path
+                self.params = result.params
+                self.query = result.query
+            except Exception as e:
+                raise urllib.error.URLError("Invalid url") from e
+
+    def _make_request(self, url=None):
+        """
+        Make github requests
+        """
+        if not url:
+            url = self.req_url
+        authorization = f'token {self.token}'
+        headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": authorization,
+        }
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            return json.loads(response.text)
+        else:
+            if "API rate limit exceeded for" in response.text:
+                raise RateLimitException
+
+    def _get_job_dir_url(self, repo):
+        """
+        List github directory
+        """
+        text = self._make_request(self.git_base_url.format(
+            repo.get('repo_name')))
+        for i in text['tree']:
+            if i['path'] in ZUUL_JOBS_LOCATION:
+                return i
+        return None
+
+    def get_file_list(self, repo):
+        """
+        Get files in directory
+        """
+        files_json = self._get_job_dir_url(repo)
+        if files_json:
+            zuul_d_files = self._make_request(files_json['url'])
+            files_list = []
+            if zuul_d_files:
+                for f in zuul_d_files['tree']:
+                    if f['path'].endswith(".yaml") or \
+                            f['path'].endswith(".yml"):
+                        files_list.append((f['path'], f['url']))
+            return files_list
+
+    def _decode_files(self, repo):
+        files_dict = {}
+        file_names = self.get_file_list(repo)
+        for (name, url) in file_names:
+            response = self._make_request(url)
+            content = response['content']
+            decoded_bytes = base64.b64decode(content)
+            decoded_str = str(decoded_bytes, "utf-8")
+            files_dict[name] = decoded_str
+        return files_dict
+
+    def get_jobs(self):
+        """
+        This method will get jobs and print
+        :return:
+        """
+        t = Tenant(name="openstack")
+        for repo in self.repos:
+            files_dict = self._decode_files(repo)
+            for key, value in files_dict.items():
+                LOG.debug("Reading: {}".format(key))
+                try:
+                    jobs_data = yaml.load(value, Loader=get_loader())
+                except Exception as e:
+                    print(e)
```

Comment:
Log this error instead of printing it into console. Let's safe 'print' just for Cibyl's final result.

Reply:
Done.